### ISSUE TITLE
Can't load a serialized predictor and utilize the GPU

### ISSUE THREAD

**Issue by lotsoffood** [awslabs/gluon-ts#313]
## Description
I trained a model with `ctx='GPU'`, and saved the resulting `predictor`. Using directions from the documentation, I get the error described in https://github.com/awslabs/gluon-ts/issues/213. 

If I run the code mentioned in the above issue, the `predictor` loads, but only predicts using the CPU. Setting `ctx='gpu'` in the code below results in the same error, and therefore I'm unable to save a trained model and load it back while being able to retain the ability to predict with the GPU.

```
predictor = RepresentableBlockPredictor.deserialize(Path("models/"))
predictor.ctx = mx.Context('cpu')
```

## To Reproduce
Same as described here: https://github.com/awslabs/gluon-ts/issues/213

## Error Message
`RuntimeError: Parameter 'deeparpredictionnetwork0_featureembedder0_cat_0_embedding_weight' was not initialized on context gpu(0). It was only initialized on [cpu(0)].`

Any thoughts on being able to save out a `predictor` object, and load it back in and still be able to utilize `GPU` acceleration for inference?

**Comment by vafl** [comment]
Can you try

```python
with mx.Context(mx.gpu(0)):
    predictor = Predictor.deserialize(Path(model_dir))
```

**Comment by lotsoffood** [comment]
This works - `INFO:root:Using GPU` when the model is loaded, and it looks like it is using the GPU for predictions. Thank you!

**Comment by lostella** [comment]
Closing since this was also properly addressed in #318 -- @lotsoffood feel free to ask further question, or reopen the issue if the problem persists!
